includes/class-wporg-gp-translation-events-active-events-cache.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+class WPORG_GP_Translation_Events_Active_Events_Cache_Entry {
+	public int $event_id;
+	public DateTimeImmutable $start;
+	public DateTimeImmutable $end;
+
+	public function __construct( int $event_id, DateTimeImmutable $start, DateTimeImmutable $end ) {
+		$this->event_id = $event_id;
+		$this->start    = $start;
+		$this->end      = $end;
+	}
+}
+
+class WPORG_GP_Translation_Events_Active_Events_Cache {
+	public const CACHE_DURATION = 60 * 60 * 24; // 24 hours.
+	private const KEY = 'translation-events-active-events';
+	private static bool $invalidated_on_event_save = false;
+
+	public function __construct() {
+		// Invalidate cache when events are modified.
+		add_filter(
+			'update_post_metadata',
+			function ( $ignore, int $object_id, string $meta_key ) {
```

Comment:
I am worried about this filter. Can't we just add an explicit invalidation after we updated the post? We control that part of the code and it's not used elsewhere. Here we're also not checking the post_type of the object.

Reply:
I agree this should be done explicitly, will change it 👍 

The reason I did it this way was mostly because of the fact that events can be edited through wp-admin as well. So we'll also need to invalidate the cache in that case. I opened an issue about that: https://github.com/WordPress/wporg-gp-translation-events/issues/40